File: ui/web-artist-mk3/src/store/orm/instrument-meme.js
/*
 * Copyright (c) XJ Music Inc. (https://xj.io) All Rights Reserved.
 */
import { Model, fk, attr } from "redux-orm";

class InstrumentMeme extends Model {
    toString() {
        return `InstrumentMeme: ${this.name}`;
    }
    // Declare any static or instance methods you need.
}
InstrumentMeme.modelName = 'InstrumentMeme';

// Declare your related fields.
InstrumentMeme.fields = {
    id: attr(),
  name: attr(),
  instrumentId: fk({
    to: 'Instrument',
    as: 'instrument',
    relatedName: 'memes',
  }),
};

export default InstrumentMeme;

File: engine/src/content/InstrumentMeme.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include "xjmusic/content/InstrumentMeme.h"

using namespace XJ;

std::set<std::string> InstrumentMeme::getNames(const std::set<const InstrumentMeme *> &instrumentMemes) {
  std::set<std::string> names;
  for (const auto &instrumentMeme: instrumentMemes) {
    names.insert(instrumentMeme->name);
  }
  return names;
}

